src/graph/Edge.java

```diff
@@ -0,0 +1,38 @@
+package graph;
+
+public class Edge {
+
+  private Vertex start;
```

Comment:
I would not recommend Edge to rely on Vertex, just use Vertex id. Our dependency path is Graph -> vertex -> edge, if you let Edge contains vertex, we will have a loop back from edge to vertex. Normal this kind of loop in dependency of classes should be avoided.